File: AEDS/Java/Trabalho Agenda/src/Agenda/Registro.java
package Agenda;

import java.util.ArrayList;
import java.util.List;

public class Registro {

    private Pessoa pessoa;
    private String nome;
    private List<Telefone> telefone = new ArrayList<Telefone>();
    private List<Endereco> enderecos = new ArrayList<Endereco>();

    public String getNumeroIdentificacao() { return this.pessoa.CadastroPessoa; }

    public String getNome() { return this.nome; }

    public String getPessoa() { return this.pessoa.getNumero(); }

    public void setPessoa(Pessoa pessoa) { this.pessoa = pessoa; }

    public void setNome(String nome) { this.nome = nome; }

    public void adicionaTelefone(String telefone){

        Telefone novoitem = new Telefone(telefone);
        this.telefone.add(novoitem);
    }

    public void removeTelefone(String telefone){

        Telefone comparador = new Telefone(telefone);

        for (Telefone i : this.telefone){

            if (i.getNumero().equals(comparador.getNumero())){

                this.telefone.remove(comparador);
                break;
            }
        }


    }

    public String[] obtemTelefones(){

        String[] retorna = new String[this.telefone.size()];

        for (int i = 0; i < this.telefone.size(); i++){

            retorna[i] = this.telefone.get(i).getNumero();

        }

        return retorna;
    }

    public void adicionaEndereco(String logradouro, int numero, String complemento,
                                 String bairro, String cidade, String estado, String CEP){

        Endereco novoItem = new Endereco(logradouro,numero,complemento,bairro,
                                         cidade,estado,CEP);

        this.enderecos.add(novoItem);
    }

    public void removeEndereco(int ID){

        for (Endereco i : this.enderecos){

            if (i.getId() == ID){

                this.enderecos.remove(i);
                break;
            }
        }
    }

    public EnderecoContainer[] obtemEnderecos(){

        EnderecoContainer[] retorno = new EnderecoContainer[this.enderecos.size()];

        for (int i = 0; i < this.enderecos.size(); i++) {

            retorno[i] = new EnderecoContainer(this.enderecos.get(i).getId(),
                    this.enderecos.get(i).getLogradouro(),this.enderecos.get(i).getNumero(),
                    this.enderecos.get(i).getComplemento(),this.enderecos.get(i).getBairro(),
                    this.enderecos.get(i).getCidade(),this.enderecos.get(i).getEstado(),
                    this.enderecos.get(i).getCEP());
        }

        return retorno;
    }

    public void Imprimir(){

        System.out.println("Nome: " + this.nome);
        System.out.println("CPF/CNPJ: " + this.pessoa.getNumero());
        System.out.println("\t----Telefones-----");
        for (Telefone i : this.telefone){
            System.out.println(i.getNumero());
        }
        System.out.println("\t----Endereços-----");
        for (Endereco i : this.enderecos){

            System.out.println("Id: " + i.getId());
            System.out.println("Logradouro: " + i.getLogradouro());
            System.out.println("Numero: " + i.getNumero());
            System.out.println("Complemento: " + i.getComplemento());
            System.out.println("Bairro: " + i.getBairro());
            System.out.println("Cidade: " + i.getCidade());
            System.out.println("Estado: " + i.getEstado());
            System.out.println("CEP: " + i.getCEP());
            System.out.println();
        }
    }
}

File: AEDS/C#/Agenda em CSharp/Registro.cs
using System;
using System.Collections.Generic;
namespace Agenda_em_CSharp
{
    class Registro
    {

        private Pessoa pessoa;
        private string Nome;
        private List<Telefone> telefones = new List<Telefone>();
        private List<Endereco> enderecos = new List<Endereco>();

        public string GetNumeroDeIdentificacao() { return this.pessoa.GetNumero(); }

        public string GetNome() { return this.Nome; }

        public void SetNome(string nome) { this.Nome = nome; }

        public void AdicionaTelefone( string Telefone){

            Telefone novoItem = new Telefone(Telefone);
            this.telefones.Add(novoItem);

        }

        public void RemoveTelefone( string Telefone) {

            bool ObjRemovido = false;

            foreach (Telefone i in this.telefones){

                if (i.Equals(Telefone)){
                    telefones.Remove(i);
                    ObjRemovido = !ObjRemovido;
                    break;
                }
            }

            if (ObjRemovido == false) Console.WriteLine("Item não encontrado");
        }

        public string[] ObtemNumeros(){

            string[] retorno = new string[this.telefones.Count];

            int i = 0;

            foreach(Telefone j in this.telefones){

                retorno[i] = j.getTelefone();
                i++;
            }

            return retorno;
        }

        public void AdicionaEndereço(String logradouro, int numero, String complemento,
                                 String bairro, String cidade, String estado, String CEP) {

            Endereco novoItem = new Endereco(logradouro, numero, complemento, bairro,
                                             cidade, estado, CEP);

            this.enderecos.Add(novoItem);
        }

        public void RemoveEndereco(int ID) {

            bool ObjRemovido = false;

            foreach( Endereco i in enderecos){

                if (i.GetId().Equals(ID)){

                    this.enderecos.Remove(i);
                    ObjRemovido = true;
                    break;
                }
            }

            if (ObjRemovido == false) Console.WriteLine(" Item não encontrado");
        }

        public EnderecoContainer[] ObtemEndereco(){

            EnderecoContainer[] retorno = new EnderecoContainer[this.enderecos.Count];

        }
    }
}
